### PR TITLE
feat(ONYX-1168): paginate the home view

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [ONYX-1168] 

### Description

This PR adds pagination to the new home view

Please be mindful that the skeletons **will** change, I didn't factor that in mind in this PR

https://github.com/user-attachments/assets/e312a702-a484-4d6c-849f-57cf1c35399c

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- paginate the home view - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-1168]: https://artsyproduct.atlassian.net/browse/ONYX-1168?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (paginate the home view - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6247beb2896aee6fddef61c784068d2f6bb41e60
</p>

File: src/app/Scenes/HomeView/HomeView.tsx [hunk 1]
```diff
@@ -19,6 +28,14 @@ export const HomeView: React.FC = () => {
             return <Section section={item} />
           }}
           ItemSeparatorComponent={() => <Spacer y={2} />}
+          onEndReached={() => loadNext(1)}
```
Comment (dblandin) [inline src/app/Scenes/HomeView/HomeView.tsx:31]: **suggestion**: What do you think of an initial fetch of 5 sections (as you currently have) but loading the next 10 sections as we scroll down?

**Review by dblandin — DISMISSED** [review]
Awesome 👍🏼 

Thanks for keeping this PR tight and focused! 💯 

One small suggestions but otherwise I think this is good to merge.

File: src/app/Scenes/HomeView/HomeView.tsx (hunk 1 above)
Comment (dblandin) [reply src/app/Scenes/HomeView/HomeView.tsx]: Bumped to this to 10 in the latest commit.

**Review by dblandin — APPROVED** [review]

**Pull request merged**